lib/chai/core/assertions.js

```diff
@@ -534,7 +535,7 @@ module.exports = function (chai, _) {
     var val = flag(this, 'object')
       , itemsCount;
 
-    switch (_.type(val)) {
+    switch (_.type(val).toLowerCase()) {
```

Comment:
Same thoughts as above.
What if instead of this `.toLowerCase()` we updated the cases, to the capitalized version?
Like you did in the switch case below, at line 1407